gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -323,6 +323,19 @@ public GoogleCloudStorageImpl(
         downscopedAccessTokenFn);
   }
 
+  private static HttpRequestInitializer setGcsRequestTracker(
+      GoogleCloudStorageOptions options, Credential credential) {
+    HttpRequestInitializer httpRequestInitializer =
+        new RetryHttpInitializer(credential, options.toRetryHttpInitializerOptions());
+    gcsRequestsTracker =
+        new GcsioTrackingHttpRequestInitializer(httpRequestInitializer, httpStatistics);
+    if (options.isGrpcEnabled()) {
+      return new RetryHttpInitializer(
+          gcsRequestsTracker, credential, options.toRetryHttpInitializerOptions());
+    }
+    return (HttpRequestInitializer) gcsRequestsTracker;
```

Comment:
please simplify. we do not need a delegate within gcsio tracker. 
it is duplicated here

Reply:
Updated